### PR TITLE
[dag] broadcast CertifiedNodeMsg with LedgerInfo

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR enables broadcasting latest ledger info with the CertifiedNode for state syncing. It obtains the ledger info from the storage adapter.

### Test Plan

Fixed unit tests.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #9830** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9830" target="_blank">Graphite</a> 
    * **PR #9831** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9831" target="_blank">Graphite</a> 
      * **PR #9724** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9724" target="_blank">Graphite</a> 
        * **PR #9968** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9968" target="_blank">Graphite</a>  👈
        * **PR #9967** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9967" target="_blank">Graphite</a> 
        * **PR #9832** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9832" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/9968?utm_source=stack-comment).

File: consensus/src/dag/dag_driver.rs [hunk 1]
```diff
@@ -151,12 +155,19 @@ impl DagDriver {
         let signature_builder =
             SignatureBuilder::new(node.metadata().clone(), self.epoch_state.clone());
         let cert_ack_set = CertificateAckState::new(self.epoch_state.verifier.len());
+        let latest_ledger_info = self
```
Comment (zekun000) [inline consensus/src/dag/dag_driver.rs:158]: carrying a db here is a bit ugly, I was thinking about caching one in the notifier adapter and rely on callback to update it, wdyt

File: consensus/src/dag/dag_handler.rs [hunk 2]
```diff
@@ -92,7 +92,7 @@ impl NetworkHandler {
                 .map(|r| r.into()),
             DAGMessage::CertifiedNodeMsg(node) => node
                 .verify(&self.epoch_state.verifier)
-                .and_then(|_| self.dag_driver.process(node))
+                .and_then(|_| self.dag_driver.process(node.certified_node()))
```
Comment (sasha8) [inline consensus/src/dag/dag_handler.rs:95]: It is a bit confusing that the nod has certified_node that has a node...
Maybe change the name of the variable?

**Review by sasha8 — APPROVED** [review]
LGTM

File: consensus/src/dag/dag_driver.rs (hunk 1 above)
Comment (ibalajiarun) [reply consensus/src/dag/dag_driver.rs:158]: i think we anyhow need the StorageAdapter here right? we can just use that instead. we can cache within the storage adapter if necessary?

Comment (zekun000) [reply consensus/src/dag/dag_driver.rs:158]: sounds better, just expose a function on the DAGStorage?

**Review by zekun000 — APPROVED** [review]

**Pull request merged**